File: src/exercises/chapter1/lesson12.rb
# frozen_string_literal: true

# S(n) = x + x^2 + x^3 + … + x^n
# @param {Numeric} x - The base value (can be a real number or integer)
# @param {Integer} n - The exponent limit (non-negative integer)
# @return {Numeric} - The sum of the series x^1 + x^2 + x^3 + … + x^n

def solution(x, n)
  result = 0

  (0..n).each do |i|
    result += x**i
  end
  result
end

File: src/exercises/chapter1/Lesson12.ts
/**
 * Lesson 12: Calculate S(n) = x + x^2 + x^3 + … + x^n
 */

export default class Lesson12 {
  /**
   * Calculates the sum S(n) = x + x^2 + x^3 + … + x^n
   *
   * @param {number} x - The base number
   * @param {number} n - The highest exponent
   * @return {number} - The sum of the series
   */
  solution(x: number, n: number): number {
    let result = 0

    // Loop from 1 to n to calculate each power of x and add it to the result
    for (let i = 1; i <= n; i++) {
      result += Math.pow(x, i) // Add x^i to the cumulative sum
    }

    return result // Return the computed sum
  }
}
